controllers/drplacementcontrol_controller.go

```diff
@@ -1125,18 +1125,21 @@ func getPlacementRule(ctx context.Context, k8sclient client.Client,
 		return nil, err
 	}
 
-	scName := usrPlRule.Spec.SchedulerName
-	if scName != RamenScheduler {
-		return nil, fmt.Errorf("placementRule %s does not have the ramen scheduler. Scheduler used %s",
-			usrPlRule.Name, scName)
-	}
+	// If either DRPC or PlacementRule are being deleted, disregard everything else.
+	if !isBeingDeleted(drpc, usrPlRule) {
```

Comment:
`ownPlacementOrPlacementRule` MAY attempt to add annotation to the PlacementObj if not already present, to own the Placement by this DRPC. If it is being deleted, then this annotation addition will fail consistently.

Technically, the PlacementObj would also not have a ramen finalizer (as that is added post the annotation addtion in the first reconcile), and hence eventually be deleted and NotFound.

Reply:
1. It's not safe to rely on the order of deletion, as it might work with the UI but fail with other tools.
2. Deletion of `PlacementDecision` is already handled, as it will be garbage collected when `DRPC` is deleted.
3. So, if `Placement` is being deleted, we don't need to worry about the annotation or anything else.
4. Similarly, if `DRPC` is being deleted, we don't need to worry about the `Placement` containing the annotation or not.

Therefore, let's fix it to avoid any potential issues.

Reply:
The only caveat here is that if the annotation was removed from the `Placement` in order to disable DR before deleting DRPC, then that might cause contention with the `Placement` on `PlacementDecision`. So in order to disable DR, DRPC needs to be deleted first.

Reply:
> 1. It's not safe to rely on the order of deletion, as it might work with the UI but fail with other tools.

Order is for all tools, not UI/console specific.

> 
>     2. Deletion of `PlacementDecision` is already handled, as it will be garbage collected when `DRPC` is deleted.

If we just want to disable DR protection, we should leave the PlacementDecision as is and not garbage collect it (which would be DRPC deletion). Not directly related to this PR, but a consideration.

> 
>     3. So, if `Placement` is being deleted, we don't need to worry about the annotation or anything else.

Then we should possibly add a check that the PlacementObj is not being deleted and then add the annotation?

> 
>     4. Similarly, if `DRPC` is being deleted, we don't need to worry about the `Placement` containing the annotation or not.
> 
> 
> Therefore, let's fix it to avoid any potential issues.

Reply:
The decision to be order independent is useful, so that there is no issue overall.

1. Placement deleted:
    - DRPC is "finalized"
    - DRPC garbage collects the PlDecision if we are still the scheduler?
        - What happens if we are not the scheduler, should we still garbage collect the PlDecision? There are no users of the same so maybe feasible.
2. DRPC is deleted:
    - DRPC is "finalized"
    - PlDecision is left as is (but I wonder who will own it in the future and delete the same, I am unsure at present)

The order of removal of the annotation is important only when disabling, not when deleting the Placement, as deletion should not change the scheduler. As a result confused what the best way forward is... still thinking.

Reply:
We may have overlooked the purpose of this PR and its intended solution. It is simply saying this:
1. If the user deleted the `Placement`, then the expectation is that the DRPC will also be deleted.
2. If DRPC is deleted and the Placement is not, the PlacementDecision will be garbage collected since it was created by DRPC.

In either case, no need to check for the annotation/scheduler. In all other cases, the check is enforced.

Additionally, If the user wants to disable DR, then the order is to delete DRPC first. Once the deletion is successful, the user proceeds with removing the annotation/scheduler from the Placement.

This rule has always been there. It is simple to understand. Ans when it is followed, there would be no window of contention.

Reply:
Merging this change, it avoids DRPC end garbage collection getting stuck on the specified user flow.